syncmaster/worker/handlers/db/base.py

```diff
@@ -44,6 +44,27 @@ def read(self) -> DataFrame:
                 name=hwm_name,
                 expression=self.transfer_dto.strategy.increment_by,
             )
+            if self.transfer_dto.resources.max_parallel_tasks > 1:
+                reader_params["options"] = {
+                    "partition_column": self.transfer_dto.strategy.increment_by,
+                    "num_partitions": self.transfer_dto.resources.max_parallel_tasks,
+                    "partitioning_mode": "range" if self.hwm and self.hwm.value else "hash",
+                }
+
+        elif self.transfer_dto.strategy.type == "full" and self.transfer_dto.resources.max_parallel_tasks > 1:
+            schema_reader = DBReader(
+                connection=self.connection,
+                table=self.transfer_dto.table_name,
+                columns=self._get_columns_filter_expressions(),
+                options={"fetchsize": 0},
+            )
+            df = schema_reader.run()
```

Comment:
nit: actually, you can call `schema = self.connection.get_df_schema(source=self.transfer_dto.table_name, columns=self._get_columns_filter_expressions())`